gradle-baseline-java/src/main/groovy/com/palantir/baseline/tasks/RefasterCompileTask.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * (c) Copyright 2019 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.baseline.tasks;
+
+import com.google.common.collect.ImmutableList;
+import java.io.File;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.gradle.api.file.ConfigurableFileCollection;
+import org.gradle.api.file.RegularFileProperty;
+import org.gradle.api.tasks.InputFiles;
+import org.gradle.api.tasks.OutputFile;
+import org.gradle.api.tasks.compile.JavaCompile;
+
+public class RefasterCompileTask extends JavaCompile {
+
+    private final ConfigurableFileCollection refasterSources = getProject().files();
+    private final RegularFileProperty refasterRulesFile = newInputFile();
```

Comment:
I think using newInputFile() results in deprecation warnings... can we use `project().objects().property(File.class)` instead?